runtime/src/main/java/org/corfudb/runtime/view/AddressSpaceView.java

```diff
@@ -268,6 +264,17 @@ ILogData read(long address) {
      * @return A result, which be cached.
      */
     public Map<Long, ILogData> read(Iterable<Long> addresses) {
+        return read(addresses, true);
+    }
+
+    /**
+     * Read the given object from a range of addresses.
+     *
+     * @param addresses An iterable with addresses to read from
+     * @param waitForWrite Flag whether wait for write is required or hole fill directly.
+     * @return A map of read addresses, which will be cached if caching is enabled
+     */
+    public Map<Long, ILogData> read(Iterable<Long> addresses, boolean waitForWrite) {
         if (!runtime.getParameters().isCacheDisabled()) {
```

Comment:
`waitForWrite` is a hack, don't expose it as a part of the `AddressSpaceView` API. We should also clean that up,

Reply:
As discussed, leaving this for now.